### PR TITLE
Implemented to parse properly regarding resource bundle usage of qml files

### PR DESCRIPTION
**Pull request by gouniLee**
The webOS app qml applications. I create this module to extract the localization string properly in qml files. Here's API spec & example. [Internal survey collab page](http://collab.lge.com/main/display/WEBOSINTL/Work%3A+%5B27%5D+QML+Localization?src=contextnavpagetreemode)
```
string qsTr(sourceText, string disambiguation, int n)
string qsTrNoOp(sourceText, string disambiguation)
QT_TR_NOOP(sourceText)
QT_TR_N_NOOP(sourceText)
stringqsTranslate(context, string sourceText, string disambiguation, int n)
string qsTranslateNoOp(context, string sourceText, string disambiguation)
QT_TRANSLATE_NOOP(context, sourceText)
QT_TRANSLATE_NOOP3(context, sourceText, disambiguation)
QT_TRANSLATE_N_NOOP(context, sourceText)
```

Reference page:
* https://doc.qt.io/qt-5/qtquick-internationalization.html
* https://doc.qt.io/qt-5/qtquick-internationalization.html#1-use-qstr-for-all-literal-user-interface-strings
* https://doc.qt.io/qt-5/qml-qtqml-qt.html#qsTr-method
* https://doc-snapshots.qt.io/qt5-5.14/linguist-id-based-i18n.html

### CODE REVIEW

File: QMLFile.js [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    var trimComment = commentString.
+        replace(/\s*\*\//, "").
+        replace(/\s*\:\s*/, "").
+        replace(/\/\s*\**\s*/, "").
+        replace(/\s*\*\s*/, "").
+        trim();
+    return trimComment;
+}
+
+var reqsTrString = new RegExp(/\b(qsTr|qsTrNoOp|QT_TR_NOOP|QT_TR_N_NOOP)\(\s*"((\\"|[^"])*)"\s*\)/g);
```
Comment (ehoogerbeets) [inline QMLFile.js]: Does QML support 'single quote' strings?

File: QMLFileType.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+ *
+ * @param {String} pathName path to the file being questions
+ * @returns {boolean} true if the path is a qml file, or false
+ * otherwise
+ */
+QMLFileType.prototype.handles = function(pathName) {
+    logger.debug("QMLFileType handles " + pathName + "?");
+    var ret = false;
+    if ((pathName.length > 4 && pathName.substring(pathName.length - 4) === ".qml") ||
+        (pathName.length > 3 && pathName.substring(pathName.length - 3) === ".js")) {
```
Comment (ehoogerbeets) [inline QMLFileType.js]: JS files can contain qml code?

File: QMLFileType.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+};
+
+/**
+ * Return the name of the node module that implements the resource file type, or
+ * the path to a QML file that implements the resource filetype.
+ * @returns {Function|undefined} node module name or path, or undefined if this file type does not
+ * need resource files
+ */
+QMLFileType.prototype.getResourceFileType = function() {
+    //return JsonResourceFileType;
```
Comment (ehoogerbeets) [inline QMLFileType.js]: Does QML not have resource file types?

File: test/testfiles/t1.qml [hunk 4]
```diff
@@ -0,0 +1,121 @@
+import QtQuick 2.6
```
Comment (ehoogerbeets) [inline test/testfiles/t1.qml]: This looks like real code. Is it okay to make it open-source? (Or is it already open-source?)

File: QMLFileType.js (hunk 2 above)
Comment (gouniLee) [reply QMLFileType.js]: We have cases of using `qsTr` and `qsTranslate` in js file in QML apps. and it should be extracted that case and need to write to the ts file.

File: QMLFileType.js (hunk 3 above)
Comment (gouniLee) [reply QMLFileType.js]: It should be `TSResourceFileType`. I will fix it.

File: test/testfiles/t1.qml (hunk 4 above)
Comment (gouniLee) [reply test/testfiles/t1.qml]: I think it's better to be simplified. I'll update it.

**Comment by gouniLee** [comment]
It has `ilib-loctool-webos-ts-resource` module dependency. but it has not been published to npm yet. That's why travis is failed

**Pull request merged**